frontend/src/metabase/lib/utils.ts

```diff
@@ -0,0 +1,216 @@
+import { t } from "ttag";
+import _ from "underscore";
+
+const LAYOUT_PROPS = [
+  "m",
+  "ml",
+  "mr",
+  "mt",
+  "mb",
+  "mx",
+  "my",
+  "p",
+  "pl",
+  "pr",
+  "pt",
+  "pb",
+  "px",
+  "py",
+  "bg",
+  "color",
+  "hover",
+  "bordered",
+];
+
+const EMAIL_REGEX =
+  /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+export function stripLayoutProps(props: unknown) {
+  return _.omit(props, LAYOUT_PROPS);
+}
+
+function s4() {
+  return Math.floor((1 + Math.random()) * 0x10000)
+    .toString(16)
+    .substring(1);
+}
+
+export function isEmpty(str: string | null) {
+  if (str != null) {
+    str = String(str);
+  } // make sure 'str' is actually a string
+  return str == null || 0 === str.length || str.match(/^\s+$/) != null;
+}
+
+// pretty limited.  just does 0-9 for right now.
+export function numberToWord(num: number) {
+  const names = [
+    t`zero`,
+    t`one`,
+    t`two`,
+    t`three`,
+    t`four`,
+    t`five`,
+    t`six`,
+    t`seven`,
+    t`eight`,
+    t`nine`,
+  ];
+
+  if (num >= 0 && num <= 9) {
+    return names[num];
+  } else {
+    return "" + num;
+  }
+}
+
+export function uuid() {
+  return (
+    s4() +
+    s4() +
+    "-" +
+    s4() +
+    "-" +
+    s4() +
+    "-" +
+    s4() +
+    "-" +
+    s4() +
+    s4() +
+    s4()
+  );
+}
+
+export function isUUID(uuid: unknown) {
```

Comment:
For this and `isBase64` and `isJWT` - is there any reason we couldn't just type the params as `string` and allow the typescript checker to ensure that we're using the functions with strings? For example if we're checking something like
```
...some other place
isUUID(235124)
```
I'd rather have the type checker catch it rather than finding out that there's an issue at runtime. I think it would make these functions look a little nicer too

Obviously there's the case of using these functions in `js` files but if we're slowly moving towards `ts/tsx` then I think it would be useful to use the checker

Reply:
I'll merge it as it is to not keep the pr open too much, but I'll add that better typing to my "small task todo"